src/Microsoft.Sbom.Extensions.DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -55,20 +55,29 @@ public static IServiceCollection AddSbomConfiguration(this IServiceCollection se
                 inputConfiguration.ToConfiguration();
                 return inputConfiguration;
             })
-            .AddSbomTool(logLevel);
+            .AddSbomTool();
         return services;
     }
 
-    public static IServiceCollection AddSbomTool(this IServiceCollection services, LogEventLevel logLevel = LogEventLevel.Information)
+    public static IServiceCollection AddSbomTool(this IServiceCollection services, ILogger? logger = null)
     {
         services
             .AddSingleton<IConfiguration, Configuration>()
-            .AddTransient(_ => FileSystemUtilsProvider.CreateInstance(CreateLogger(logLevel)))
-            .AddTransient(x =>
+            .AddSingleton(x =>
             {
-                logLevel = x.GetService<InputConfiguration>()?.Verbosity?.Value ?? logLevel;
-                return Log.Logger = CreateLogger(logLevel);
+                if (logger != null)
+                {
+                    return logger;
+                }
+                else
+                {
+                    var level = x.GetService<InputConfiguration>()?.Verbosity?.Value ?? LogEventLevel.Information;
+                    var defaultLogger = CreateDefaultLogger(level);
+                    Log.Logger = defaultLogger;
+                    return defaultLogger;
+                }
```

Comment:
This change was made because when I tried to pass in an optional ILogger, the separate transient ILogger resolution that was side-effecting `Log.Logger` seemed to cause an infinite loop. Since there's only one ILogger, making these a) singleton, and b) doing the Log.Logger overriding only for cases where the logger wasn't explicitly known beforehand seems to solve the problem.  The tool-driven use case shouldn't be effected here.